### PR TITLE
Bookstore: add reducers and actions

### PR DESCRIPTION
**Pull request by batoorsohail**
### In this project, we added the followings:

- Redux Toolkit
-  A directory that contains all Redux logic
- Redux store
- Two slices
- Reducer that adds a book
- Reducer that removes a book
- Reducer that checks the status

### CODE REVIEW

File: src/redux/books/booksSlice.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  initialState,
+  reducers: {
+    addBook: (state, actions) => {
+      const bookData = actions.payload;
+      state.books.push(bookData);
+    },
+    removeBook: (state, actions) => {
+      const idOfBookToRemove = actions.payload.id;
+      state.books.filter((book) => book.id !== idOfBookToRemove);
+    },
```
Comment (codepantha) [inline src/redux/books/booksSlice.js]: - [ ] This operation returns the filtered books. Please save the result of the `Array.prototype.filter()` method in `state.books`.
**HINT:** 
```js
state.books = state.books.filter((book) => book.id !== idOfBookToRemove);
```

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED ♻️ 

## Hi @batoorsohail 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights 🎯 
✔️ Descriptive PR
✔️ Configure redux store 📌 
✔️ Write actions and reducers 📌
✔️ No linter errors

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- n/a

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/books/booksSlice.js (hunk 1 above)
Comment (batoorsohail) [reply src/redux/books/booksSlice.js]: Thanks @codepantha 🙌

**Review by OLIPLICHE — APPROVED** [review]
Hi @batoorsohail, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Highlights
- Applied the change requested by the review👏 👏
- Linters is green👏 👏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**